File: problem26/p26.java
public class p26 {
	
	public static int reciprocalCycles(int d){
	    int i, length = 0, remainder, pos, res = 0;

	    for(i=1; i<d; i++){
	    	int[] remainders = new int[i+1];
	        remainder = 1;
	        pos = 0;
	        while(remainders[remainder] == 0 && remainder != 0){
	            remainders[remainder] = pos;
	            remainder *= 10;
	            remainder %= i;
	            pos++;
	        }
	        if(pos - remainders[remainder] > length){
	            length = pos - remainders[remainder];
	            res = i;
	        }
	    }

	    return res;
	}

	public static void main(String[] args) {

		System.out.println(reciprocalCycles(1000));
		
	}

}

File: problem26/p26.py
def reciprocal_cycles(d):
    res = 0
    length = 0
    for i in range(1, d):
        remainders = [0] * d
        remainder = 1
        pos = 0
        while remainders[remainder] == 0 and remainder != 0:
            remainders[remainder] = pos
            remainder *= 10
            remainder %= i
            pos += 1
        if pos - remainders[remainder] > length:
            length = pos - remainders[remainder]
            res = i
    return res, length


print(reciprocal_cycles(1000)[0])
